src/NServiceBus.NHibernate.Tests/TimeoutPersister/When_removing_timeouts_from_the_storage.cs

```diff
@@ -79,30 +81,97 @@ public void Should_remove_timeouts_by_id()
         }
 
         [Test]
-        public void Should_remove_timeouts_by_sagaid()
+        public void TryRemove_should_return_false_when_timeout_already_deleted()
+        {
+            var timeout = new TimeoutData();
+
+            persister.Add(timeout);
+
+            Assert.IsTrue(persister.TryRemove(timeout.Id));
```

Comment:
How is this even possible? The timeout being stored has no ID assigned. Is it assigned by the database?